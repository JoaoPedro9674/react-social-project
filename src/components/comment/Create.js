import PropTypes from 'prop-types';
import React from 'react';

class CreateComment extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            comment: ''
        };
        this.handleCommentUpdate = this.handleCommentUpdate.bind(this);
    }
    handleCommentUpdate(event) {
        this.setState({
            comment: event.target.value
        });
    }
    handleSubmit(event) {
        event.preventDefault();
        const { threadID, handleSubmit } = this.props;
        console.log('Submitting');
        console.log(event.target);
        handleSubmit(threadID);
    }
    render() {
        return (
            <form onSubmit={this.handleSubmit} className="create-comment">
                <input
                    type="text"
                    placeholder="Write a comment..."
                    onChange={this.handleCommentUpdate}
                    className="create-comment"
                />
            </form>
        );
    }
}

CreateComment.propTypes = {
    threadID: PropTypes.string.isRequired,
    handleSubmit: PropTypes.func.isRequired
};

export default CreateComment;
