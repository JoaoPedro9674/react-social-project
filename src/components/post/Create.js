import React, { PropTypes } from 'react';

class CreatePost extends React.Component {
  constructor(props) {
    super(props);

    // Set up state
    this.state = {
      content: '',
    };

    // Set up event handlers
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handlePostChange = this.handlePostChange.bind(this);
  }

  handlePostChange(event) {
    const content = event.target.value;
    this.setState({ content });
  }

  handleSubmit(event) {
    event.preventDefault();
    if (this.state.content.length <= 0) {
      return;
    }
    if (this.props.onSubmit) {
      const newPost = {
        content: this.state.content,
      };
      this.props.onSubmit(newPost);
      this.setState({
        content: '',
      });
    }
  }

  render() {
    return (
      <form className="create-post" onSubmit={this.handleSubmit}>
        <textarea value={this.state.content} onChange={this.handlePostChange} placeholder="What's on your mind?" />
        <button className="btn btn-default">Post</button>
      </form>
    );
  }
}

CreatePost.propTypes = {
  onSubmit: PropTypes.func,
};

export default CreatePost;
