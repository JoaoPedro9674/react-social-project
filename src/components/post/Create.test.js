import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';

import { CreatePost } from './Create';

describe('<Create/>', () => {
    let sandbox;
    beforeEach(() => {
        sandbox = sinon.sandbox.create();
    });
    afterEach(() => {
        sandbox.restore();
    });
    describe('constructor', () => {
        it('should be constructed properly', () => {
            const onSubmitStub = sandbox.stub();
            const component = shallow(<CreatePost onSubmit={onSubmitStub}/>);
            expect(component.instance().handleSubmit).toBeDefined();
            expect(component.instance().handlePostChange).toBeDefined();
        });
    });
    describe('event methods', () => {
        it('should handle a post submission', () => {
            const onSubmitStub = sandbox.stub();
            const preventDefaultStub = sandbox.stub();
            const wrapper = shallow(<CreatePost onSubmit={onSubmitStub}/>);
            const mockNewValue = 'new value';
            wrapper.find('textarea').simulate('change', {
                target: {
                    value: mockNewValue
                }
            });
            wrapper.find('form').simulate('submit', {
              preventDefault: preventDefaultStub
            });

            expect(onSubmitStub.calledOnce).toBe(true);
            expect(preventDefaultStub.calledOnce).toBe(true);
            const calledWithArgs = onSubmitStub.args[0][0];
            expect(calledWithArgs.id).toBeDefined();
            expect(calledWithArgs.date).toBeDefined();
            expect(calledWithArgs.content).toBe(mockNewValue);
        });
    });
    describe('render methods', () => {
        it('should render correctly', () => {
            const onSubmitStub = sandbox.stub();
            const wrapper = shallow(<CreatePost onSubmit={onSubmitStub}/>);
            expect(wrapper.find('form').length).toBe(1);
            expect(wrapper.find('textarea').length).toBe(1);
            expect(wrapper.find('button').length).toBe(1);
        });
    });
});
