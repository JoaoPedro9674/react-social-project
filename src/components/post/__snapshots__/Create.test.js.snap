exports[`<Create/> render methods should render correctly 1`] = `
<form
  className="create-post"
  onSubmit={[Function]}>
  <textarea
    onChange={[Function]}
    placeholder="What\'s on your mind?"
    value="" />
  <button
    className="btn btn-default">
    Post
  </button>
</form>
`;
